File: frontend/src/types/incident.ts
export type Severity = 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW';

export type IncidentType = 
  | 'MANUAL_SOS'
  | 'POSSIBLE_FALL'
  | 'PROLONGED_IMMOBILITY'
  | 'ROUTE_DEVIATION'
  | 'GEOFENCE_BREACH';

export type NetworkStatus = 'ONLINE' | 'WEAK' | 'OFFLINE';

export type SyncStatus = 'PENDING' | 'SYNCED' | 'CONFLICT';

export type IncidentStatus = 
  | 'DETECTED'
  | 'ASSESSED'
  | 'LOCATING'
  | 'AWAITING_RESPONSE'
  | 'TEAM_ASSIGNED'
  | 'RESCUE_DISPATCHED'
  | 'TEAM_ON_SITE'
  | 'RESOLVED';

export interface GPSLocation {
  latitude: number;
  longitude: number;
}

export interface EstimatedLocation {
  latitude: number;
  longitude: number;
  confidence: number;
  radiusMeters: number;
}

export interface MovementPoint {
  latitude: number;
  longitude: number;
  timestamp: string;
  speedMs?: number;
}

export interface TimelineEntry {
  status: string;
  timestamp: string;
  actor: string;
  note?: string;
}

export interface EmergencyCapsule {
  id: string;
  incidentId: string;
  touristId: string;
  timestamp: string;
  lastKnownLocation: GPSLocation;
  estimatedLocation: EstimatedLocation;
  movementHistory: MovementPoint[];
  batteryLevel: number;
  networkStatus: NetworkStatus;
  syncStatus: SyncStatus;
  incidentType: IncidentType;
  severity: Severity;
  emergencyContacts: string[];
  checksum: string;
  encryptedPayload: string;
}

export interface Incident {
  incidentId: string;
  touristId: string;
  incidentType: IncidentType;
  severity: Severity;
  timestamp: string;
  lastKnownLocation: GPSLocation;
  estimatedLocation: EstimatedLocation;
  movementHistory: MovementPoint[];
  batteryLevel: number;
  networkStatus: NetworkStatus;
  syncStatus: SyncStatus;
  status: IncidentStatus;
  assignedTeamId?: string;
  assignedTeamName?: string;
  etaMinutes?: number;
  emergencyContacts: string[];
  timeline: TimelineEntry[];
}

export interface RescueTeam {
  id: string;
  name: string;
  type: 'Forest Rescue' | 'Police' | 'Medical' | 'Disaster Response';
  latitude: number;
  longitude: number;
  available: boolean;
  equipment: string[];
  baseLocation: string;
  activeIncidentId?: string;
  estimatedResponseTime: number;
  distanceKm?: number;
  calculatedEtaMinutes?: number;
}

export type ViewRole = 'CONTROL_ROOM' | 'TOURIST_PWA' | 'RESCUE_TEAM' | 'DEMO_GUIDE';

File: backend/app/schemas/incident.py
from pydantic import BaseModel, Field
from typing import Optional, List, Any
from enum import Enum

class IncidentTypeEnum(str, Enum):
    MANUAL_SOS = "MANUAL_SOS"
    POSSIBLE_FALL = "POSSIBLE_FALL"
    PROLONGED_IMMOBILITY = "PROLONGED_IMMOBILITY"
    ROUTE_DEVIATION = "ROUTE_DEVIATION"
    GEOFENCE_BREACH = "GEOFENCE_BREACH"

class SeverityEnum(str, Enum):
    CRITICAL = "CRITICAL"
    HIGH = "HIGH"
    MEDIUM = "MEDIUM"
    LOW = "LOW"

class NetworkStatusEnum(str, Enum):
    ONLINE = "ONLINE"
    WEAK = "WEAK"
    OFFLINE = "OFFLINE"

class SyncStatusEnum(str, Enum):
    PENDING = "PENDING"
    SYNCED = "SYNCED"
    CONFLICT = "CONFLICT"

class StatusEnum(str, Enum):
    DETECTED = "DETECTED"
    ASSESSED = "ASSESSED"
    LOCATING = "LOCATING"
    AWAITING_RESPONSE = "AWAITING_RESPONSE"
    TEAM_ASSIGNED = "TEAM_ASSIGNED"
    RESCUE_DISPATCHED = "RESCUE_DISPATCHED"
    TEAM_ON_SITE = "TEAM_ON_SITE"
    RESOLVED = "RESOLVED"

class GPSLocation(BaseModel):
    latitude: float
    longitude: float

class EstimatedLocation(BaseModel):
    latitude: float
    longitude: float
    confidence: float = 85.0
    radiusMeters: float = 250.0

class MovementPoint(BaseModel):
    latitude: float
    longitude: float
    timestamp: str
    speedMs: Optional[float] = 0.0

class TimelineEntry(BaseModel):
    status: str
    timestamp: str
    actor: str
    note: Optional[str] = None

class IncidentCreate(BaseModel):
    incidentId: str
    touristId: str
    incidentType: str
    severity: str
    timestamp: str
    lastKnownLocation: GPSLocation
    estimatedLocation: EstimatedLocation
    movementHistory: Optional[List[dict]] = []
    batteryLevel: int = 100
    networkStatus: str = "ONLINE"
    syncStatus: str = "PENDING"
    status: str = "DETECTED"
    assignedTeamId: Optional[str] = None
    assignedTeamName: Optional[str] = None
    emergencyContacts: Optional[List[str]] = []
    timeline: Optional[List[dict]] = []

class IncidentStatusUpdate(BaseModel):
    status: str
    actor: str = "Control Room"
    note: Optional[str] = None

class TeamAssignmentRequest(BaseModel):
    teamId: str
    actor: str = "Control Room Admin"

class IncidentResponse(BaseModel):
    incidentId: str
    touristId: str
    incidentType: str
    severity: str
    timestamp: str
    lastKnownLocation: GPSLocation
    estimatedLocation: EstimatedLocation
    movementHistory: List[dict] = []
    batteryLevel: int
    networkStatus: str
    syncStatus: str
    status: str
    assignedTeamId: Optional[str] = None
    assignedTeamName: Optional[str] = None
    etaMinutes: Optional[int] = None
    emergencyContacts: List[str] = []
    timeline: List[dict] = []

    class Config:
        from_attributes = True
